app/workers/travis_build_killer.rb

```diff
@@ -0,0 +1,35 @@
+class TravisBuildKiller
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+  include MiqToolsServices::SidekiqWorkerMixin
+  sidekiq_options :queue => :miq_bot, :retry => false
+
+  recurrence { hourly.minute_of_hour(0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55) }
+
+  def perform
+    if !first_unique_worker?
+      logger.info "#{self.class} is already running, skipping"
+    else
+      kill_builds
+    end
+  end
+
+  private
+
+  def kill_builds
+    repo = CommitMonitorRepo.where(:upstream_user => "ManageIQ", :name => "manageiq").first
+    repo.with_travis_service do |travis|
+      builds_to_cancel = travis.builds
+        .take_while { |b| b.pending? || b.canceled? }
+        .reject(&:canceled?)
```

Comment:
@Fryguy I don't understand this, why are we taking pending and canceled builds and then rejecting canceled?